include/zim/archive.h

```diff
@@ -108,16 +110,29 @@ namespace zim
        *
        *  Construct an archive from a descriptor of a file with an embedded ZIM
        *  archive inside.
+       *  Fd is used only at Archive creation.
+       *  Ownership of the fd is not taken and it must be closed by caller.
        *
        *  Note: This function is not available under Windows.
        *
-       *  @param fd The descriptor of a seekable file with a continuous segment
-       *  representing a complete ZIM archive.
-       *  @param offset The offset of the ZIM archive relative to the beginning
-       *  of the file (rather than the current position associated with fd).
-       *  @param size The size of the ZIM archive.
+       *  @param fd A FdInput (tuple) containing the fd (int), offset (offset_type) and size (size_type)
+       *            referencing a continuous segment representing a complete ZIM archive.
        */
-      Archive(int fd, offset_type offset, size_type size);
+      explicit Archive(FdInput fd);
+
+      /** Archive constructor.
+       *
+       *  Construct an archive from several file descriptors.
+       *  Each part may be embedded in a file.
+       *  Fds are used only at Archive creation.
+       *  Ownership of the fds is not taken and they must be closed by caller.
+       *
+       *  Note: This function is not available under Windows.
+       *
+       *  @param fds A vector of FdInput (tuple) containing the fd (int), offset (offset_type) and size (size_type)
+       *             referencing a series of segments representing a complete ZIM archive.
+       */
```

Comment:
I think a separate note will be useful on the situation when multiple segments belong to the same file. As far as I understand, providing access to them via the same shared fd is allowed.

Reply:
Done